web-common/src/features/dashboards/state-managers/selectors/dimension-filters.ts

```diff
@@ -21,22 +25,86 @@ export const selectedDimensionValues = (
     // if it is a complex filter unsupported by UI then no values are selected
     if (isExpressionUnsupported(dashData.dashboard.whereFilter)) return [];
 
+    const dimExpr = getWhereFilterExpression(dashData)(dimName);
+    if (
+      dimExpr?.cond?.op &&
+      (dimExpr.cond.op === V1Operation.OPERATION_LIKE ||
+        dimExpr.cond.op === V1Operation.OPERATION_NLIKE)
+    )
+      return [];
+
     // FIXME: it is possible for this way of accessing the filters
     // to return the same value twice, which would seem to indicate
     // a bug in the way we're setting the filters / active values.
     // Need to investigate further to determine whether this is a
     // problem with the runtime or the client, but for now wrapping
     // it in a set dedupes the values.
-    return [
-      ...new Set(
-        getValuesInExpression(
-          getWhereFilterExpression(dashData)(dimName),
-        ) as string[],
-      ),
-    ];
+    return [...new Set(getValuesInExpression(dimExpr) as string[])];
   };
 };
 
+export const selectedDimensionValuesV2 = (
+  instanceId: string,
+  metricsViewNames: string[],
+  whereFilter: V1Expression | undefined,
+  dimensionName: string,
+  timeStart?: string,
+  timeEnd?: string,
+): CompoundQueryResult<string[]> => {
+  // if it is a complex filter unsupported by UI then no values are selected
+  if (!whereFilter || isExpressionUnsupported(whereFilter))
+    return readable({
+      isFetching: false,
+      isLoading: false,
+      error: undefined,
+      data: [],
+    });
+
+  const dimExpr = whereFilter.cond?.exprs?.find((e) =>
+    matchExpressionByName(e, dimensionName),
+  );
+  if (!dimExpr?.cond?.op)
+    return readable({
+      isFetching: false,
+      isLoading: false,
+      error: undefined,
+      data: [],
+    });
```

Comment:
Is this a type guard? Or should this return an error?

Reply:
This is the case when a dimension has no filter. It should return empty array similar to how it was before.